vmdb/app/models/miq_provision_request_template.rb

```diff
@@ -1,32 +1,15 @@
 class MiqProvisionRequestTemplate < MiqProvisionRequest
-
   def create_tasks_for_service(service_task, parent_svc)
-
     template_service_resource = ServiceResource.find_by_id(service_task.options[:service_resource_id])
     scaling_min = template_service_resource.nil? ? 1 : template_service_resource.scaling_min
 
-    tasks = []
-    0.upto(scaling_min - 1).each do |idx|
-
-      task = self.create_request_task(idx)
-      task.options[:miq_force_unique_name] = [true, 1]
-      task.options[:service_guid] = parent_svc.guid
-      task.options[:service_resource_id] = template_service_resource.id
-      task.options[:service_template_request] = false
-      task.userid = service_task.userid
-      user = User.find_by_userid(task.userid)
-      unless user.nil?
-        task.options[:owner_email]      = user.email
-        task.options[:owner_first_name] = user.first_name
-        task.options[:owner_last_name]  = user.last_name
+    0.upto(scaling_min - 1).collect do |idx|
+      create_request_task(idx) do |req_task|
+        udpate_service_options(req_task, parent_svc, template_service_resource)
+        update_owner(req_task, service_task)
+        req_task.miq_request_id = service_task.miq_request.id
```

Comment:
@kbrock The reason we moved away from pushing req_task onto the miq_request_tasks list is that it caused the req_task to be saved as part of that call instead of waiting to save until after the block ends.